lib/modules/manager/pep621/schema.ts

```diff
@@ -83,6 +84,15 @@ export const PyProjectSchema = z.object({
       'build-backend': z.string().optional(),
     })
     .optional(),
+  'dependency-groups': z
+    .record(
+      z.string(),
+      z
+        .array(z.union([z.string(), z.object({})]))
+        // Skip non-string entries, like `{include-group = "typing"}`, as they are not dependencies.
+        .transform((deps) => deps.filter(is.string)),
```

Comment:
Rather use the `LooseArray` utility

```suggestion
      // Skip non-string entries, like `{include-group = "typing"}`, as they are not dependencies.
      LooseArray(z.string())
```